HACKING.md

```diff
@@ -149,10 +149,26 @@ which will output a URL to connect to with a browser, such as
 <http://localhost:8765/qunit/base1/test-dbus.html>. Adjust the path for different
 tests and inspect the results there.
 
-You can also run individual tests by specifying the `TESTS` environment
-variable:
+QUnit tests are run as part of a pytest test called `test_browser`.  You can
+run individual tests via `pytest -k`, like so:
 
-    make check TESTS=qunit/base1/test-chan.html
+    pytest -k test-fsinfo.html
+
+You can see JavaScript code coverage information for QUnit tests.  For a
+summary table:
+
+    pytest -k test_browser --js-cov
```

Comment:
Is there anything that stops us from calling that in our pytest-cov workflow? That would be more insightful than the line number lists in "Missing". Or is it too much output?

Reply:
I think the answer to this question is the above question ;)

Reply:
I tested this on this branch, but with `pytest -k test_browser --js-cov` it looks exactly as before:
```
======================================================== JavaScript coverage summary ========================================================
Path                                                     % (bytes covered) ~ uncovered lines
pkg/base1/test-base64.js                             91.8% (1921 / 2092) ~ 15-18, 35-38, 66-69
pkg/base1/test-cache.js                              98.7% (5354 / 5424) ~ 72-74
[...]
```
i.e. just line numbers. So that's indeed what we already do. It would be interesting to have a mode where it runs *all* unit tests and shows missing coverage -- that'd be especially useful for things like cockpit.js as its various parts are covered by different unit tests. I naïvely tried

    pytest -k test_browser --js-cov-files='*'

and that works fine.